scheduler/predicate.go

```diff
@@ -4,39 +4,40 @@ import (
 	"encoding/json"
 	"net/http"
 	"strconv"
+	"strings"
 
 	"github.com/cybozu-go/topolvm"
 	corev1 "k8s.io/api/core/v1"
 )
 
-func filterNodes(nodes corev1.NodeList, requested int64) ExtenderFilterResult {
-	if requested <= 0 {
+func filterNodes(nodes corev1.NodeList, requested map[string]int64) ExtenderFilterResult {
+	if len(requested) <= 0 {
 		return ExtenderFilterResult{
 			Nodes: &nodes,
 		}
 	}
 
-	required := uint64(requested)
 	filtered := corev1.NodeList{}
 	failed := FailedNodesMap{}
 
+NODE_LOOP:
 	for _, node := range nodes.Items {
```

Comment:
I think it would be better to make the procedures inside this a separate function,
and call it as a goroutine for better readability and performance.